metrics/internal/collectors/storageconsumer.go

```diff
@@ -0,0 +1,99 @@
+package collectors
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/metrics/internal/options"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+var _ prometheus.Collector = &StorageConsumerCollector{}
+
+type StorageConsumerCollector struct {
+	Informer                cache.SharedIndexInformer
+	StorageConsumerMetadata *prometheus.Desc
+	AllowedNamespace        string
```

Comment:
I think this is unused.

Reply:
It has been used in a lot of places for the `Collect` , `Describe` and `Run` functions

Reply:
Can you please point me there?

Reply:
Let's keep it as a follow up. There are some cases to be considered here regarding namespaces.